frontend/app/src/components/staking/liquity/LiquityStatistics.vue

```diff
@@ -0,0 +1,348 @@
+const { assetPrice } = useBalancePricesStore();
+const LUSD_ID = 'eip155:1/erc20:0x5f98805A4E8be255a32880FDeC7F6728C6568bA0';
+const lusdPrice = assetPrice(LUSD_ID);
+
+const { tc } = useI18n();
+
+const { isLoading } = useStatusStore();
+const loading = isLoading(Section.DEFI_LIQUITY_STATISTICS);
+
+const current: Ref<boolean> = ref(false);
+
+const statisticWithAdjustedPrice: ComputedRef<LiquityStatistic | null> =
+  computed(() => {
+    const statisticVal = get(statistic);
+
+    if (!statisticVal) {
+      return null;
+    }
+
+    if (!get(current)) {
+      return statisticVal;
+    }
+
+    const stakingGains = statisticVal.stakingGains.map(
+      (stakingGain: AssetBalance) => {
+        const price = get(assetPrice(stakingGain.asset)) ?? One;
+
+        return {
+          ...stakingGain,
+          usdValue: stakingGain.amount.multipliedBy(price)
+        };
+      }
+    );
+
+    const stabilityPoolGains = statisticVal.stabilityPoolGains.map(
+      (stabilityPoolGain: AssetBalance) => {
+        const price = get(assetPrice(stabilityPoolGain.asset)) ?? One;
+
+        return {
+          ...stabilityPoolGain,
+          usdValue: stabilityPoolGain.amount.multipliedBy(price)
+        };
+      }
+    );
+
+    return {
+      ...statisticVal,
+      totalDepositedStabilityPoolUsdValue:
+        statisticVal.totalDepositedStabilityPool.multipliedBy(
+          get(lusdPrice) ?? One
+        ),
+      totalWithdrawnStabilityPoolUsdValue:
+        statisticVal.totalWithdrawnStabilityPool.multipliedBy(
+          get(lusdPrice) ?? One
+        ),
+      stakingGains,
+      stabilityPoolGains
+    };
+  });
+
+const totalPnl: ComputedRef<BigNumber | null> = computed(() => {
+  const statisticVal = get(statistic);
+  const poolVal = get(pool);
+
+  if (!statisticVal || !poolVal) {
+    return null;
+  }
+
+  // A = Total Deposited Stability Pool - Total Withdrawn Stability Pool
+  const expectedAmount = statisticVal.totalDepositedStabilityPool.minus(
+    statisticVal.totalWithdrawnStabilityPool
+  );
+
+  // Liquidations Gains (current price)
+  const liquidationGainsInCurrentPrice = poolVal.gains.amount.multipliedBy(
+    get(assetPrice(poolVal.gains.asset)) ?? One
+  );
+
+  // Rewards (current price)
+  const rewardsInCurrentPrice = poolVal.rewards.amount.multipliedBy(
+    get(assetPrice(poolVal.rewards.asset)) ?? One
+  );
+
+  // B = Total Gains Stability Pool + Liquidations Gains (current price) + Rewards (current price)
+  const totalWithdrawals = statisticVal.totalUsdGainsStabilityPool
+    .plus(liquidationGainsInCurrentPrice)
+    .plus(rewardsInCurrentPrice);
+
+  // C = A - Current deposited amount
+  const diffDeposited = expectedAmount.minus(poolVal.deposited.amount);
+
+  // convert C to USD at current price
+  const diffDepositedInCurrentUsdPrice = diffDeposited.multipliedBy(
+    get(lusdPrice) ?? One
+  );
+
+  // PnL = B - C
+  return totalWithdrawals.minus(diffDepositedInCurrentUsdPrice);
```

Comment:
this seems like a formula for calculation, can you extract it to a normal method (no ref passing) and convert the comments to a jsdoc description of what the method does?